python/whylogs/api/logger/__init__.py

```diff
@@ -33,8 +40,22 @@ def log(
     row: Optional[Dict[str, Any]] = None,
     schema: Optional[DatasetSchema] = None,
     name: Optional[str] = None,
+    multiple: Optional[Dict[str, Loggable]] = None,
 ) -> ResultSet:
-    return TransientLogger(schema=schema).log(obj, pandas=pandas, row=row, name=name)
+    if multiple is not None:
+        result_sets: Dict[str, ResultSet] = {}
+        for alias, d in multiple.items():
+            result_set = log(obj=d, schema=schema)
+            result_sets[alias] = result_set
+
+        # Return one result set with everything in it since we have to return result_sets
+        result_set = reduce(lambda r1, r2: r1.merge(r2), result_sets.values())
+        notebook_session_log_comparison(multiple, result_sets)
+        return result_set
+    else:
+        result_set = TransientLogger(schema=schema).log(obj, pandas=pandas, row=row, name=name)
+        notebook_session_log(result_set, obj, pandas=pandas, row=row, name=name)
```

Comment:
I wonder if there is an easy high level flag or something we can pull up for a check for which path to execute? I'm looking for a very easy to reason about structure on if the code is executing notebook onboarding flow vs production logging. The `mulitple is not None` is good in that way, but I have to step through notebook_session_log to know if its early exiting or doing something interesting?

Reply:
makes sense, maybe an updated name helps clarify.

I was looking for an easy way to avoid a potential future edit to `notebook_session_log` interfering with a production use case's why.log when there is no init called. I was considering putting some flag variable there and wrapping this all in an if block something like:
`if session_is_defined():`